examples/summarize_scoring_using_bedrock.ipynb

```diff
@@ -0,0 +1,374 @@
+    "{}\n",
+    "\"\"\"\n",
+    "\n",
+    "def generate_summary_from_llama(model_id, article): \n",
+    "    body = json.dumps({\"prompt\": prompt.format(article)})\n",
+    "    modelId = model_id\n",
+    "    accept = \"application/json\"\n",
+    "    contentType = \"application/json\"\n",
+    "    \n",
+    "    response = bedrock_runtime.invoke_model(\n",
+    "        body=body, modelId=modelId, accept=accept, contentType=contentType\n",
+    "    )\n",
+    "    response_body = json.loads(response.get(\"body\").read())\n",
+    "    return response_body.get(\"generation\")\n",
+    "\n",
+    "\n",
+    "def generate_summary_from_titan(model_id, article):\n",
+    "    body = json.dumps({\"inputText\": prompt.format(article)})\n",
+    "    modelId = model_id\n",
+    "    accept = \"application/json\"\n",
+    "    contentType = \"application/json\"\n",
+    "    \n",
+    "    response = bedrock_runtime.invoke_model(\n",
+    "        body=body, modelId=modelId, accept=accept, contentType=contentType\n",
+    "    )\n",
+    "    response_body = json.loads(response.get(\"body\").read())\n",
+    "    return response_body.get(\"results\")[0].get(\"outputText\")\n",
+    "\n",
+    "\n",
+    "def generate_summary_from_claude(model_id, article):\n",
+    "    body = json.dumps({\n",
+    "        \"prompt\": f\"Human:\\n{prompt.format(article)}\\nAssistant:\\n\",\n",
+    "        \"max_tokens_to_sample\": 4000,  # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-claude.html\n",
+    "    })\n",
+    "    modelId = model_id\n",
+    "    accept = \"application/json\"\n",
+    "    contentType = \"application/json\"\n",
+    "    \n",
+    "    response = bedrock_runtime.invoke_model(\n",
+    "        body=body, modelId=modelId, accept=accept, contentType=contentType\n",
+    "    )\n",
+    "    response_body = json.loads(response.get(\"body\").read())\n",
+    "    return response_body.get(\"completion\")    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4f76d25c-7241-48db-9c82-e00335d6da29",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "models = (\n",
+    "    \"meta.llama2-13b-chat-v1\",\n",
+    "    \"meta.llama2-70b-chat-v1\",\n",
+    "    \"amazon.titan-text-lite-v1\",\n",
+    "    \"anthropic.claude-v2\",\n",
+    "    \"anthropic.claude-v1\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ea41fa3a-13c8-4c07-89f5-6b5e2e58edcb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from collections import defaultdict\n",
+    "summaries = defaultdict(list)\n",
+    "summaries[\"input\"] = articles"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b94c7974-410e-4c7c-b213-9555cc8c0871",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for model in [m for m in models if m not in summaries.keys()]:\n",
```

Comment:
think this example code would be simpler if we did fewer of these operations in a for loop (looks like you looped through all models here and made sure the cell was re-runnable by only non-redundant summaries by checking in the summaries dict)

Reply:
notebook is ok as is